app/viewmodels/helpers/ItemsAddToListHelper.scala

```diff
@@ -0,0 +1,155 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.helpers
+
+import controllers.sections.items.routes
+import models.requests.{CnCodeInformationItem, DataRequest}
+import models.sections.items.ItemsAddToListItemModel
+import models.{Index, NormalMode}
+import pages.sections.items.{ItemBulkPackagingChoicePage, ItemCommodityCodePage, ItemExciseProductCodePage, ItemsSectionItem}
+import play.api.i18n.Messages
+import play.twirl.api.HtmlFormat
+import queries.{ItemsCount, ItemsPackagingCount}
+import services.GetCnCodeInformationService
+import uk.gov.hmrc.govukfrontend.views.Aliases.Text
+import uk.gov.hmrc.govukfrontend.views.viewmodels.content.HtmlContent
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist._
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.Logging
+import viewmodels.checkAnswers.sections.items.{ItemBrandNameSummary, ItemCommercialDescriptionSummary, ItemPackagingSummary, ItemQuantitySummary}
+import viewmodels.govuk.TagFluency
+import viewmodels.govuk.summarylist._
+import viewmodels.taskList.{Completed, InProgress}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+
+class ItemsAddToListHelper @Inject()(tag: views.html.components.tag,
+                                     span: views.html.components.span,
+                                     cnCodeInformationService: GetCnCodeInformationService,
+                                     itemPackagingSummary: ItemPackagingSummary) extends TagFluency with Logging {
+
+
+  def allItemsSummary(implicit request: DataRequest[_], messages: Messages, headerCarrier: HeaderCarrier, ec: ExecutionContext): Future[Seq[SummaryList]] =
+    getAddedItems match {
+      case Nil => Future.successful(Nil)
+      case items => itemsWithUnitOfMeasure(items).map(_.map(summaryList))
+    }
+
+  private def itemsWithUnitOfMeasure(items: Seq[ItemsAddToListItemModel])
+                                    (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext): Future[Seq[ItemsAddToListItemModel]] =
+    cnCodeInformationService.getCnCodeInformation(CnCodeInformationItem(items).distinct).map { response =>
+      items.map { item =>
+        item.copy(unitOfMeasure = response.collectFirst {
+          case (cnItem, cnInfo) if cnItem.productCode == item.exciseProductCode && item.commodityCode.contains(cnItem.cnCode) =>
+            cnInfo.unitOfMeasure
+        })
+      }
+    }
+
+  private def getAddedItems(implicit request: DataRequest[_]): Seq[ItemsAddToListItemModel] =
+    request.userAnswers.get(ItemsCount).fold(Seq.empty[ItemsAddToListItemModel]) { count =>
```

Comment:
Again, personal preference: 
```suggestion
    request.userAnswers.get(ItemsCount).fold[Seq[ItemsAddToListItemModel]](Seq.empty) count =>
```